File: 812.最大三角形面积.cpp
/*
 * @lc app=leetcode.cn id=812 lang=cpp
 *
 * [812] 最大三角形面积
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    double largestTriangleArea(vector<vector<int>>& points) {
        double res = 0;
        for (auto && p1 : points) {
            for (auto && p2 : points) {
                for (auto && p3 : points) {
                    double x1 = p1[0];
                    double x2 = p2[0] - x1;
                    double x3 = p3[0] - x1;
                    double y1 = p1[1];
                    double y2 = p2[1] - y1;
                    double y3 = p3[1] - y1;
                    double tmp = abs(x2 * y3 - x3 * y2) / 2;
                    if (tmp > res)
                        res = tmp;
                }
            }
        }
        return res;
    }
};
// @lc code=end

File: 812.最大三角形面积.go
/*
 * @lc app=leetcode.cn id=812 lang=golang
 *
 * [812] 最大三角形面积
 */
package leetcode

import "math"

// @lc code=start
func largestTriangleArea(points [][]int) float64 {
	var res float64
	for i := 0; i < len(points); i++ {
		for j := 0; j < len(points); j++ {
			for k := 0; k < len(points); k++ {
				x1 := float64(points[i][0])
				x2 := float64(points[j][0]) - x1
				x3 := float64(points[k][0]) - x1
				y1 := float64(points[i][1])
				y2 := float64(points[j][1]) - y1
				y3 := float64(points[k][1]) - y1
				if tmp := math.Abs(x2 * y3 - x3 * y2) / 2; tmp > res {
					res = tmp
				}
			}
		}
	}
	return res
}
// @lc code=end

